### PR TITLE
icf: Updated regexes.

### PR DESCRIPTION
**Pull request by JakubNvk**
Reviewer: @pepol 
Issue: #1

### CODE REVIEW

File: parser/ICF.java [hunk 1]
```diff
@@ -20,48 +20,47 @@
    */
   Pattern RE_CONTINUED_LINE = Pattern.compile("^.*[^\\\\]\\\\$");
 
-  // Regex that matches line containing only comment (starting with '#' or ';').
+  /* Regex that matches line containing only comment (starting with '#' or
+   *';').
```
Comment (pepol) [inline parser/ICF.java]: Why this change? The original line fits into the 80 character limit.

File: parser/ICF.java [hunk 2]
```diff
@@ -20,48 +20,47 @@
...
+   */
   Pattern RE_COMMENT_LINE = Pattern.compile("^[#;].*$");
+  
+  /* Value utility regexes. These should be appended behind the actual
```
Comment (pepol) [inline parser/ICF.java]: Please note that multiline comments should look like this:
```java
/*
 * This is a multiline comment.
 *
 * Also it can contain multiple paragraphs separated by a blank line.
 */
```

File: parser/ICF.java [hunk 3]
```diff
@@ -20,48 +20,47 @@
...
-   * - decimal - values (integer and floating point numbers).
-   */
   Pattern RE_NUMBER_VALUE = Pattern.compile(
       "((?:0x[0-9a-fA-F]+)|(?:0b[01]+)|(?:0[0-7]+)|(?:[0-9]*(?:\\.[0-9]+)?))");
-
-  // Regex that matches string enclosed in double quotes ("").
   Pattern RE_QUOTED_STRING = Pattern.compile("\"((?:(?:\\\\\")|[^\"])*)\"");
-
-  // Regex that matches unquoted string.
   Pattern RE_UNQUOTED_STRING = Pattern.compile("((?:(?:\\\\\")|[^\"#;])*)");
```
Comment (pepol) [inline parser/ICF.java]: The `RE_UNQUOTED_STRING` should also start matching from the beginning of the line.

**Comment by pepol** [comment]
@JakubNvk generally looks great, but please address my comments.

File: parser/ICF.java (hunk 1 above)
Comment (JakubNvk) [inline parser/ICF.java]: It was 81 characters actually.

File: parser/ICF.java (hunk 2 above)
Comment (JakubNvk) [inline parser/ICF.java]: @pepol yeah, forgot to edit those properly, my bad.

File: parser/ICF.java (hunk 1 above)
Comment (pepol) [inline parser/ICF.java]:
```bash
> wc
// Regex that matches line containing only comment (starting with '#' or ';').
      1      13      79
```

Comment (pepol) [inline parser/ICF.java]: Oh sorry, I didn't see it was inside a class, so 2 more for spaces... okay then.

Comment (JakubNvk) [inline parser/ICF.java]: Well it's showing 81 for me.. :D

**Comment by JakubNvk** [comment]
@pepol ping

**Comment by pepol** [comment]
LGTM, please squash.
